iceoryx_hoofs/source/cli/option.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/internal/cli/option.hpp"
+
+namespace iox
+{
+namespace cli
+{
+namespace internal
+{
+bool Option::isSwitch() const noexcept
+{
+    return value.empty();
+}
+
+bool Option::isEmpty() const noexcept
+{
+    return longOption.empty() && shortOption == NO_SHORT_OPTION;
+}
+
+bool Option::longOptionNameDoesStartWithDash() const noexcept
+{
+    return !longOption.empty() && longOption[0] == '-';
+}
+
+bool Option::shortOptionNameIsEqualDash() const noexcept
+{
+    return shortOption == '-';
+}
+
+bool Option::hasLongOptionName(const OptionName_t& value) const noexcept
```

Comment:
Alright but then it would be inconsistent with the other methods. But I could rename all methods to something like `matchesLongOptionName`, `matchesShortOptionName`, `matchesOptionName` but I prefer the current names.